maven/maven_functions.sh

```diff
@@ -5,16 +5,25 @@
 
 # 1st arg: action to run for each pom found in the build directory
 # 2nd arg: path to root of repo
-# 3rd arg: optional relative repo path for a more targeted pom search, must start with a '/'
+# 3rd arg: custom classifier for jar artifacts, uses the placeholder "None" to 
+#          mean string 
+# 4th arg: optional relative repo path for a more targeted pom search, must start with a '/'
 _for_each_pom() {
     action=$1
     repo_root_dir_path=$2
-    pom_root_path=$3
+    custom_jar_classifier=$3
+    pom_root_path=$4
 
     if ! [[ "$action" =~ ^(install_main_artifact|build_javadoc_jar|install_sources_and_javadoc_jars|upload_all_artifacts|clean_source_tree)$ ]]; then
         echo "ERROR: Unknown action $action" && exit 1
     fi
 
+    # a "None" value for jar_artifact_classifier means it should be unset
+    # (passing empty string arguments to bash functions is problematic)
+    if [ "$custom_jar_classifier" = "None" ]; then
```

Comment:
Shouldn't this be done in _deploy_artifacts_to_nexus? Because the classifier is not the last arg in the positional args?

Reply:
Yes, I think you are right, thanks.